src/targets/gpu/fuse_mlir.cpp

```diff
@@ -325,14 +323,24 @@ struct find_mlir_standalone_convolution_op
     }
 };
 
+struct find_mlir_standalone_convolution_op : find_mlir_standalone_op
+{
+    auto matcher() const { return match::name("convolution"); }
+};
+
+struct find_mlir_standalone_dot_op : find_mlir_standalone_op
```

Comment:
The module name need to be unique;
right now, 
```c++
module_ref mm = mpm.create_module("mlir_" + std::to_string(counter++));
```
will create name collisions for standalone convs and gemms.

So my suggestion would be make the op name a string in find_mlir_standalone_op struct and do something like : 
```c++
match::find_matches(mpm, find_mlir_standalone_op{"dot"});
```
```c++
match::find_matches(mpm, find_mlir_standalone_op{"convolution"});
```

Reply:
I know this because I tried this with resnet yesterday :)

Reply:
But, `counter` is a static and it is in the base class.  `counter` will be incremented regardless whether if was called from `find_mlir_standalone_convolution_op` or `find_mlir_standalone_dot_op`.

I also used resnet and there was no name collisions

Reply:
oh right you have the static var in the base class where as in I just copied for my experiment. So ignore my comment then.